frontend/src/scenes/plugins/edit/pluginSourceLogic.tsx

```diff
@@ -0,0 +1,154 @@
+import { actions, afterMount, kea, key, listeners, path, props, reducers, selectors } from 'kea'
+
+import type { pluginSourceLogicType } from './pluginSourceLogicType'
+import { forms } from 'kea-forms'
+import api from 'lib/api'
+import { loaders } from 'kea-loaders'
+import { lemonToast } from 'lib/components/lemonToast'
+import { validateJson } from 'lib/utils'
+import React from 'react'
+import { FormErrors } from 'lib/forms/Errors'
+
+interface PluginSourceProps {
+    id: number
+    onClose?: () => void
+}
+
+interface PluginSource {
+    name: string
+    'index.ts': string
+    'config.json': string
+}
+
+const defaultSource = `// Learn more about plugins at: https://posthog.com/docs/plugins/build/overview
+
+// Processes each event, optionally transforming it
+export function processEvent(event, { config }) {
+    // Some events (such as $identify) don't have properties
+    if (event.properties) {
+        event.properties['hello'] = \`Hello \${config.name}\`
+    }
+    // Return the event to be ingested, or return null to discard
+    return event
+}
+
+// Runs when the plugin is loaded, allows for preparing it as needed
+export function setupPlugin (meta) {
+    console.log(\`The date is \${new Date().toDateString()}\`)
+}
+
+// Runs every hour on the hour
+async function runEveryHour(meta) {
```

Comment:
aware it's not your code but putting it here given this function currently won't work given the others are exported and it isn't 

```suggestion
export async function runEveryHour(meta) {
```